sdk/storage/Azure.Storage.Common.DataMovement/src/FileSystemScanner.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Storage.Common.DataMovement
+{
+    internal class FileSystemScanner
```

Comment:
I'm not sure this should be called `FileSystemScanner` if we're pointing it to a path rather than root.

Should this be `PathScanner` ?

Or maybe we should have `IPathScanner` and implementations like `DirectoryScanner` and `FileScanner` ? But then what does it mean to scan a file?

Reply:
Fair point. Probably a PathScanner would be best, so that a caller only has to worry about coming up with the path, and the scanner does the work of figuring out whether its just one or several items. The separate implementations would make little sense (at least with the latter; FileScanner logic would pretty much amount to `File.GetAttributes(fullPath) & FileAttributes.Directory) != FileAttributes.Directory` not throwing, which probably doesn't need a whole class)

Reply:
Maybe we're doing this split onto dir/file scanners prematurely. Should we just have PathScanner that encapsulates all this logic for now and see how it evolves?
@amnguye @gapra-msft thoughts ?

Reply:
I was actually just now thinking the same thing. I think the FilesystemScanner -> PathScanner should be able to handle both files and directories. No need for a Directory/FileScanner for now